### PR TITLE
Create Models and relationships

### PR DESCRIPTION
**Pull request by thecodingpoet**
#### What does this PR do?
This PR creates models and relationships

#### Description of Task to be completed?
- Create User Model
- Create Contact Model
- Create Email Address Model
- Create Phone Number Model
- Define the relationships between each models

#### How should this be manually tested?

NA

#### Any background context you want to provide?
NA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-application up and running.
+Keep track of your contacts, all in one place, anytime, anywhere!
 
-Things you may want to cover:
+## Technologies
 
-* Ruby version
+- Ruby on Rails
+- React.js
+- Materialize CSS
```
Comment (fiyinadebayo) [inline README.md]: Really? Materialize? What's you thoughts on this, why this choice?

File: README.md [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-
-* Deployment instructions
-
-* ...
+1. Fork this repositry to your account.
+2. Clone your repository: git clone git@github.com:andela-oduyile/contact.me.git
+3. Create your feature branch: git checkout -b new-feature
+4. Commit your changes: git commit -m "did something"
+5. Push to the remote branch: git push origin new-feature
+6. Open a pull request.
```
Comment (fiyinadebayo) [inline README.md]: Update last line to `Open a pull request, ensure that the "base fork" points to the correct repository and branch.`

File: README.md [hunk 3]
```diff
@@ -1,24 +1,24 @@
-# README
+# Contact.me
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+Keep track of your contacts, all in one place, anytime, anywhere!
 
-Things you may want to cover:
+## Technologies
```
Comment (fiyinadebayo) [inline README.md:5]: What versions of the technologies listed below?

File: README.md [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+- React.js
+- Materialize CSS
 
-* System dependencies
+## Local Development
 
-* Configuration
+- Install npm dependencies bundle install
+- To run the app: rails s
+- To run the tests run: rspec
```
Comment (fiyinadebayo) [inline README.md]: `bundle exec rspec`

File: README.md [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
+- Ruby on Rails
+- React.js
+- Materialize CSS
 
-* System dependencies
+## Local Development
 
-* Configuration
+- Install npm dependencies bundle install
+- To run the app: rails s
```
Comment (fiyinadebayo) [inline README.md]: `rails server`

**Review by fiyinadebayo — APPROVED** [review]

File: db/migrate/20171016124659_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class CreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :number
```
Comment (fiyinadebayo) [inline db/migrate/20171016124659_create_users.rb]: number or mobile_number or phone_number?

File: spec/factories/contacts.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
```
Comment (fiyinadebayo) [inline spec/factories/contacts.rb:1]: You can use https://github.com/stympy/faker to generate fake 'random' test data.

File: spec/factories/phone_numbers.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :phone_number do
+    number "MyString"
+    type ""
+    contact nil
```
Comment (fiyinadebayo) [inline spec/factories/phone_numbers.rb]: You can update this to bind to the model, like so

> after(:build) do |phone_number|
>    phone_number.contact = create(:contact)
> end

in the place of `contact nil` and same in the other places.

File: spec/models/user_spec.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (fiyinadebayo) [inline spec/models/user_spec.rb:4]: I project tests shall be coming soon, right?

**Review by fiyinadebayo — APPROVED** [review]

**Comment by fiyinadebayo** [comment]
Really nice work man! @andela-oduyile

File: README.md (hunk 3 above)
Comment (thecodingpoet) [reply README.md:5]: Done

File: README.md (hunk 1 above)
Comment (thecodingpoet) [reply README.md]: I would just take it out for now till we figure out what is best

File: README.md (hunk 5 above)
Comment (thecodingpoet) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (thecodingpoet) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (thecodingpoet) [reply README.md]: Done

File: db/migrate/20171016124659_create_users.rb (hunk 6 above)
Comment (thecodingpoet) [reply db/migrate/20171016124659_create_users.rb]: Changed this to phone_number

File: spec/factories/contacts.rb (hunk 7 above)
Comment (thecodingpoet) [reply spec/factories/contacts.rb:1]: Done

File: spec/factories/phone_numbers.rb (hunk 8 above)
Comment (thecodingpoet) [reply spec/factories/phone_numbers.rb]: Oh thanks. Done

File: spec/models/user_spec.rb (hunk 9 above)
Comment (thecodingpoet) [reply spec/models/user_spec.rb:4]: Yes starting the tests for models right away

**Comment by thecodingpoet** [comment]
Implemented all feedback @ceemion

**Pull request merged**